sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/command.py

```diff
@@ -4,32 +4,46 @@
 from marshmallow import fields
 
 from azure.ai.ml._internal._schema.node import InternalBaseNodeSchema, NodeType
-from azure.ai.ml._schema import NestedField, StringTransformedEnum
-from azure.ai.ml._schema.component.retry_settings import RetrySettingsSchema
-from azure.ai.ml._schema.core.fields import DistributionField
+from azure.ai.ml._schema import (
+    NestedField,
+    StringTransformedEnum,
+    UnionField,
+    RegistryStr,
+    AnonymousEnvironmentSchema,
+    ArmVersionedStr
+)
+from azure.ai.ml._schema.job import ParameterizedParallelSchema, ParameterizedCommandSchema
 from azure.ai.ml._schema.job.job_limits import CommandJobLimitsSchema
-from azure.ai.ml._schema.job_resource_configuration import JobResourceConfigurationSchema
+from azure.ai.ml.constants._common import AzureMLResourceType
 
 
-class CommandSchema(InternalBaseNodeSchema):
+class CommandSchema(InternalBaseNodeSchema, ParameterizedCommandSchema):
+    class Meta:
+        exclude = ["code", "distribution"]  # internal command doesn't have code & distribution
+
+    environment = UnionField(
+        [
+            RegistryStr(azureml_type=AzureMLResourceType.ENVIRONMENT),
+            NestedField(AnonymousEnvironmentSchema),
+            ArmVersionedStr(azureml_type=AzureMLResourceType.ENVIRONMENT, allow_default_version=True),
+        ],
+    )
     type = StringTransformedEnum(allowed_values=[NodeType.COMMAND], casing_transform=lambda x: x)
```

Comment:
Nit: can we directly use string + validate here since no transform needed?

Reply:
There is some risk as StringTransformedEnum is different from fields.Str in error handling. Given there is already a DumpableEnumField, I will use it instead.